src/Lpt/UI/Home.hs

```diff
@@ -31,24 +36,69 @@ import           CLI                            ( User(..)
                                                 )
 import           Control.Monad.IO.Class         ( liftIO )
 import           Data.Either                    ( rights )
+import           Data.HashMap.Strict            ( elems
+                                                , keys
+                                                )
 import           Data.Text                      ( unpack )
 import           Data.Vector                    ( fromList )
 import           Graphics.Vty.Input.Events      ( Event )
-import           Parse.Types                    ( Item
-                                                , getId
+import qualified Parse.Types                   as T
+                                                ( Complex(..)
+                                                , Login(..)
+                                                , Note(..)
+                                                )
+import           Parse.Types                    ( Item(..)
                                                 , getName
                                                 )
 import           UI.Types                       ( Name(..)
                                                 , TuiState(..)
                                                 )
 
 drawHomepage :: List Name Item -> [Widget Name]
-drawHomepage = pure . border . hLimit 50 . renderList renderElement True
+drawHomepage items = pure (listWidget items <+> itemWidget items)
+ where
+  listWidget =
+    borderWithLabel (str "Items")
+      . hLimitPercent 30
+      . renderList renderElement True
+  itemWidget vaultItems =
+    borderWithLabel (label selected) . renderItemWidget $ selected
+   where
+    label = str . maybe "" (itemType . snd)
+    itemType (MkLogin   _) = "Password"
+    itemType (MkNote    _) = "Secure Note"
+    itemType (MkComplex _) = "Other"
+    selected = listSelectedElement vaultItems
 
 renderElement :: Bool -> Item -> Widget Name
-renderElement focus item =
-  highlight . vBox . map (\f -> str (f item)) $ [getId, getName]
-  where highlight = if focus then border else id
+renderElement _ item =
+  padTopBottom 1 . vBox . map (\f -> str (f item)) $ [getName]
+
+renderItemWidget :: Maybe (Int, Item) -> Widget Name
+renderItemWidget (Just (_, item)) = renderItem item
+renderItemWidget Nothing          = str "No item selected"
+
+renderItem :: Item -> Widget Name
+renderItem (MkLogin (T.Login iden name username password group url note)) =
+  infoWidget loginKeys
+    <+> vBox (map strWrap [iden, name, username, password, group, url, note])
```

Comment:
Data constructors with a bunch of string fields scare me. Are we sure these are in the correct order? Use the named fields of the `T.Login` record instead:
```suggestion
renderItem :: Item -> Widget Name
renderItem (MkLogin (T.Login {..})) =
  infoWidget loginKeys
    <+> vBox (map strWrap [_id, _name, _username, _password, _group, _url, _note])
```
or:
```suggestion
renderItem :: Item -> Widget Name
renderItem (MkLogin login) =
  infoWidget loginKeys
    <+> vBox (map strWrap [T._id login, T._name login, T._username login, T._password login, T._group login, T._url login, T._note login])
```
(I'm unsure about the syntax. I haven't programmed Haskell in years.)